### DISCUSSION TITLE
Improve process to testing PR

### DISCUSSION THREAD

**Discussion by Jean-BaptisteC** [organicmaps/organicmaps#4464] [Ideas]
Recently, i have more corrections to test on my devices.
I try to find all bugs and crash during my tests and it's not easy because, i don't know everything IHM in OM
Thanks for your trust :)
When i have somes corrections to test in same times, it's more complicated, because i have somes debug apk on my devices and i need to know which apk fix what.

Also, i want to prioritize testing on corrections waiting in next release, but it's complicated because i don't know when release is generated. (Informations fill in project are available in each PR)

I suggest to create project to follow PR testing, of course, if we create project, is to save time and more follow progression about testing.

I have create a draft project on my repo https://github.com/users/Jean-BaptisteC/projects/1/views/1 , please check and tell me, if you are interessting to create same project on OM repo
If necessary, i can give you access on my project to improve my draft project.

CC @biodranik @vng @rtsisyk

**Comment by biodranik** [comment]
Thanks for the idea! Don't you think that it complicates the process? How do you see the whole workflow, keeping in mind our limited work time?

At the moment, our master branch should be always well-tested and ready to release at any moment in time. That's why we focus on specific PRs and try to test them/find any bugs before merging them into the master branch. If PRs have issues, they're not merged and are not released. If there is an issue in the master branch, then it's prioritized if necessary and the fix PR appears.

We'll try to keep milestones more or less actual, but that also requires a bit of work and time.

**Reply by Jean-BaptisteC** [reply]
I suggest this idea for a better coordination. For me to know, if PR is waiting for next release and for you to know, if i have started my tests and progression in my tests.

On project, i suggest to do not use automation workflow (Automation is not perfect, github allow only add PR in project, when statut on PR is update, open, closed, merged):
- I suggest a statut to know, if i have started tests, if i waiting corrections, if my tests is done
- A field to know, if PR is waiting for next release
- A date field, to know when corrections are waiting (if corrections is waiting for next release)
- And a commentary field to add somes informations

Project is easy to use, because you can see directly in PR these informations

In my mind, you have just add PR in project and fill fields if necessary after, i update informations in project when my tests are started or finish

Please check my project on my repo. If you think is more complicated and take a lot of time, let's find another solution to improve process and prioritize my tests on PR waiting for release.

**Reply by biodranik** [reply]
We have a good experience with Github Projects. My questions are not about this tool. They are more about the problem we're trying to solve. What is precisely the issue that you try to optimize using projects? Will it make our work easier, not more complicated?

**Reply by Jean-BaptisteC** [reply]
Sometimes, PR are not push before you publish release, because corrections haven't been testing.
In somes cases, it's can be troublesome, when corrections fixed a crash or boring bug and community is hangry because problem is not solved and users can be write bad commentaries on stores.

I suggest to use a project to more follow tests on PR and to know which PR have need to be testing in priority.

I dont' think add project, give more work at everyone
- For maintainers, we have just to add PR in project, if you want someones tests changes and you can follow test progression directly in PR.
- For testers, we have just to update fields in project when tests are started or is stopped, because we waiting corrections.

I think it can be a real interest to follow tests on big PR (Place page refactoring like example) because in cases if we have refactoring or new features, we tests several times each time after developers have take our feedback on changes and push new commits. Of course follow little PR with changes tested in sames day is not interesting except if corrections is waiting for next release.

I hope you understood better my vision

Have a nice day :)

**Reply by biodranik** [reply]
I don't mind if this approach makes your life/work simpler ) I don't see many people willing to test our changes (actually, it's only you now). If someone else will join, it may be beneficial to track/share who is testing what.

**Reply by Jean-BaptisteC** [reply]
Yes, this project have more interest when we have somes users to test PR.
I understood your position about my idea, we keep idea for later :).
For now we can keep our functioning to test PR. But when you tag me to test PR, tell me if corrections is waiting for next release or add pr to releases blockers milestone.

**Reply by biodranik** [reply]
We'll try to keep issues/PRs in appropriate milestones, so you'll see it. Blockers are rare, most of the PRs are highly _desired_ to be in the Next Release )
